### ISSUE TITLE
if the larger --update-freq  the better?

### ISSUE THREAD

**Issue by jiezhangGt** [pytorch/fairseq#979]
if the larger ```--update-freq``` the better?  I found that when I increase the ```--update-freq```（12）,  the training is unstable, BLEU fluctuate widely（~0.6-1BLEU）

**Comment by myleott** [comment]
Increasing `--update-freq` increases the batch size by accumulating gradients from several mini-batches. Typically when training with larger batches it's important to retune the learning rate to achieve optimal performance. You might look at this paper for more analysis of this: https://arxiv.org/pdf/1806.00187.pdf. `--update-freq` corresponds to `cumul` in Table 1.

**Comment by gvskalyan** [comment]
There are 2 cumuls . can you please elaborate the difference between these?
By 128 GPUs you mean 128 in number or 16 GPU with update-freq 8 to simulate training?

**Comment by huihuifan** [comment]
You need to increase the learning rate if you are going to increase the batchsize. 128 gpus is equivalent to 16 gpu with update freq 8, assuming your batchsize/num tokens stays the same. I'm closing this issue.
